src/dashboard/webserver/server.py

```diff
@@ -0,0 +1,92 @@
+                           existing_devices=DeviceInformation.get_mac_addresses(g.db))
+
+
+def _anomaly_configuration(form_data: dict):
+    """
+    Handler for anomaly detection configuration form, updates current anomaly configuration settings used in the database
+    :param form_data: Form data submitted by the user from the webpage
+    :return: nothing
+    """
+    a = AnomalyEquations(average_equation=form_data['aequations'],
+                         adjustment_equation=form_data['sdequations'])
+    AnomalyEquations.insert_new_object(a, conn=g.db)
+
+
+def _email_configuration(form_data: dict):
+    """
+    Handler for email configuration form, updates current email settings used in the database,
+    database only stores one configuration at a time
+    :param form_data: Form data submitted by the user from the webpage
+    :return: nothing
+    """
+    e = EmailInformation(recipient_addresses=form_data['raddress'],
+                         sender_address=form_data['saddress'],
+                         sender_email_password=form_data['password'],
+                         smtp_server=form_data['server'])
+    EmailInformation.insert_new_object(e, conn=g.db)
+
+
+def _device_configuration(form_data: dict, ip_neighbors):
+    """
+    Handler for device configuration form, inserts and removes devices the user would like monitored by the IDS
+    :param form_data: Form data submitted by the user from the webpage
+    :param ip_neighbors: List of neighboring devices displayed to the user
+    :return: nothing
+    """
+    # insert unique devices into the database to be monitored by our IDS
+    mac_addresses = {key for key in form_data.keys()}
+    existing_devices = set(DeviceInformation.get_mac_addresses(g.db))
+    new_devices_macs = mac_addresses - existing_devices
+    new_devices = [neigh for neigh in ip_neighbors if neigh.mac in new_devices_macs]
+    for item in new_devices:
+        d = DeviceInformation(mac_address=item.mac,
+                              name="",
+                              ip_address=item.ip)
+        DeviceInformation.insert_new_object(d)
+    # remove devices that were unchecked
+    devices_to_delete = [DeviceInformation.get_by_pk(DeviceInformation.mac_address, mac_address, g.db) for mac_address
+                         in
+                         existing_devices - mac_addresses]
+    for device in devices_to_delete:
+        device.delete(False, g.db)
+    DeviceInformation.safe_commit(g.db)
+
+
+@app.route('/ids-priv/ids-alerts/')
+def ids_alerts():
+    all_alerts = get_alerts('IDS', g.db)
+    return render_template('alerts.htm', all_alerts=all_alerts, dashboard_title='IDS')
+
+
+@app.route('/ids-priv/privacy-alerts/')
+def privacy_alerts():
+    all_alerts = get_alerts('Privacy', g.db)
+    return render_template('alerts.htm', all_alerts=all_alerts, dashboard_title='Privacy')
+
+
+if __name__ == "__main__":
+    app.run(debug=True, host="0.0.0.0", port=5000)
```

Comment:
Two changes:

- `debug=False`
- `use_reloader=False`

Reply:
@samir-elhelw76 Please resolve

Reply:
I believe @samir-elhelw76 said this was purely for testing, so I guess it is fine to stay there